basedpyright/__main__.py

```diff
@@ -0,0 +1,11 @@
+from __future__ import annotations
+
+from basedpyright.run_node import run
+
+
+def main():
```

Comment:
If we put this into `__init__`, then we could add capability for programmatic invocation:
```py
from basedpyright import main

main("foo/bar.py", type_check_mode="strict")
```

@DetachHead opinion?

Reply:
thats something we could do in the future if someone requests it but i think it's fine as-is for now